client-c/main.cpp

```diff
@@ -256,3 +218,58 @@ void openSocket() {
   }
   std::cout << "Socket connection to K Server is open\n";
 }
+
+void countBrackets(const char *buffer, size_t len) {
+  for(int i = 0; i < len; i++) {
+    switch(buffer[i]){
+      case '{':{
+        brace_counter_++;
+        break;
+      }
+      case '}':{
+        brace_counter_--;
+        break;
+      }
+      case '[':{
+        bracket_counter_++;
+        break;
+      }
+      case ']':{
+        bracket_counter_--;
+        break;
+      }
+    }
```

Comment:
This loop can be factored out into its own function which both `countBrackets` and `doneReading` can use I think?

Reply:
Yeah, I could factor out part of the loop body.